runtime/src/accounts_db.rs

```diff
@@ -1809,16 +1923,144 @@ impl AccountsDB {
+                }
+            }
+        }
+        flush_rooted_accounts_cache_elapsed.stop();
+        inc_new_counter_info!(
+            "flush_rooted_accounts_cache",
+            flush_rooted_accounts_cache_elapsed.as_us() as usize
+        );
+    }
+
+    fn flush_slot_cache(&self, slot: Slot) {
+        info!("flush_slot_cache slot: {}", slot);
+        if let Some(slot_cache) = self.accounts_cache.slot_cache(slot) {
+            let iter_items: Vec<_> = slot_cache.iter().collect();
+            let (accounts, hashes): (Vec<(&Pubkey, &Account)>, Vec<Hash>) = iter_items
+                .iter()
+                .map(|iter_item| {
+                    let key = iter_item.key();
+                    let account = &iter_item.value().account;
+                    let hash = iter_item.value().hash;
+                    ((key, account), hash)
+                })
+                .unzip();
+
+            self.store_with_hashes(slot, &accounts, &hashes, false);
+        };
+
+        self.accounts_cache.remove_slot(slot);
+    }
+
+    fn cache_store_id() -> usize {
+        AppendVecId::MAX
+    }
+
+    fn cache_accounts(
+        &self,
+        slot: Slot,
+        hashes: &[Hash],
+        accounts_and_meta_to_store: &[(StoredMeta, &Account)],
+    ) -> Vec<AccountInfo> {
+        assert_eq!(hashes.len(), accounts_and_meta_to_store.len());
+        accounts_and_meta_to_store
+            .iter()
+            .zip(hashes)
+            .map(|((meta, account), hash)| {
+                self.accounts_cache
+                    .store(slot, &meta.pubkey, (**account).clone(), *hash);
+                AccountInfo {
+                    store_id: Self::cache_store_id(),
+                    offset: 0,
+                    lamports: account.lamports,
+                }
+            })
+            .collect()
+    }
+
+    fn store_accounts_to<F: FnMut(Slot) -> Arc<AccountStorageEntry>, P: Iterator<Item = u64>>(
+        &self,
+        slot: Slot,
+        accounts: &[(&Pubkey, &Account)],
+        hashes: &[Hash],
+        storage_finder: F,
+        mut write_version_producer: P,
+        use_cache: bool,
+    ) -> Vec<AccountInfo> {
+        let default_account = Account::default();
+        let accounts_and_meta_to_store: Vec<(StoredMeta, &Account)> = accounts
+            .iter()
+            .map(|(pubkey, account)| {
+                let account = if account.lamports == 0 {
+                    &default_account
+                } else {
+                    *account
+                };
+                let data_len = account.data.len() as u64;
+
+                let meta = StoredMeta {
+                    write_version: write_version_producer.next().unwrap(),
+                    pubkey: **pubkey,
+                    data_len,
+                };
+                (meta, account)
+            })
+            .collect();
+
+        if use_cache {
+            self.cache_accounts(slot, hashes, &accounts_and_meta_to_store)
```

Comment:
entrypoint for caching

Reply:
aka, rabbit hole. xD

Reply:
nits: how about aligning name like this?: `write_accounts_to_caches`/`write_accounts_to_stores`?

Reply:
done!